app/src/main/resources/score-tracker.js

```diff
@@ -0,0 +1,735 @@
+ * marked as complete when the user navigates to the next question after answering the last one.
+ * Therefore, the function also checks if `exerciseComplete` is false to avoid redundant completions.
+ */
+function isExerciseComplete(questionIndex, maxQuestionIndex, exerciseComplete) {
+    return (questionIndex + 1) >= maxQuestionIndex && !exerciseComplete;
+}
+
+
+/**
+ * Sends an xAPI statement indicating the completion of an exercise.
+ *
+ * @param {object} object - The activity object that represents the exercise being completed.
+ * @param {number} correctAnswers - The number of correct answers provided by the user during the exercise.
+ * @param {number} maxQuestionIndex - The total number of questions in the exercise.
+ *
+ * @returns {Promise<void>} A promise that resolves when the xAPI statement has been successfully sent.
+ *
+ */
+async function sendCompletionXAPIStatement(object, correctAnswers, maxQuestionIndex) {
+    const duration = recordProgress(startExerciseTime)
+    const scaled = correctAnswers / maxQuestionIndex
+    const passed = (scaled * 100) >= passingGrade;
+    const result = {
+        score: {
+            scaled: scaled,
+            raw: correctAnswers,
+            min: 0,
+            max: maxQuestionIndex
+        },
+        completion: true,
+        success: passed,
+        duration: duration
+    };
+    const completionXAPIStatement = createXAPIStatement(
+        "completed", object, result,
+        addParentAndGroupingToContext(xapiConfig.context) // context with parent exercise added
+    );
+    await sendXAPIStatement(completionXAPIStatement);
+}
+
+
+/**
+ * Sends an xAPI statement indicating that a question has been answered.
+ *
+ * @param {questionObject} questionObject - The activity object representing the question that was answered.
+ * @param {resultObject} result - the xapi result object of the user's response to the question
+ *
+ * @returns {Promise<void>} A promise that resolves when the xAPI statement has been successfully sent.
+ */
+async function sendQuestionXAPIStatement(questionObject, resultObject, context) {
+    const questionXAPIData = createXAPIStatement(
+        "answered", questionObject, resultObject, context
+    );
+
+    await sendXAPIStatement(questionXAPIData);
+}
+
+/**
+ * Sends an xAPI statement to the configured Learning Record Store (LRS).
+ * @param {object} xAPIData - The xAPI statement object that will be sent to the LRS.
+ * @returns {Promise<void>} A promise that resolves when the xAPI statement has been successfully sent, or rejects if an error occurs.
+ */
+async function sendXAPIStatement(xAPIData) {
+    try {
+        console.log("sending xapi data")
+        const endpointUrl = new URL(xapiConfig.endpoint);
+        // Remove any trailing slash from the path and add 'statements/' at the end
+        endpointUrl.pathname = endpointUrl.pathname.replace(/\/+$/, '') + '/statements/';
```

Comment:
If you use URL("statements", endpoint), it actually replaces /xapi instead of adding to it. That’s why I handled it by manually adding "statements" to the path — to keep /xapi/statements intact. I should have explained it better on the code

Reply:
done